src/utilities/district.ts

```diff
@@ -67,6 +67,11 @@ export function getVulnerableSeatByQuotient(districtResult: DistrictResult): Vul
     };
 }
 
+export function getPartyNameByPartyCode(districtResult: DistrictResult, partyCode: string) {
```

Comment:
This is an exceptionally slow function -- the reason for this being that "find" will read through an array in an iterative manner. A lookup of some sort would be fast. A DistrictResult being needed means that the function is tightly coupled to the model. Whereas I see this is a simple way to implement it, it is also the least reusable way of doing so. 

Ideally this function would, early on, gather the information needed and store it in state. That is at least what was discussed in the meeting. Its input should only be the party code and its output should be the party name or undefined.

Reply:
One possible solution..

district.ts
```
export function getPartyNameDict(districtResult: DistrictResult): {[partyCode: string]: string} {
    const partyNameDict: {[characterName: string]: string} = {};
    districtResult.partyResults.forEach((d) =>{  
        partyNameDict[d.partyCode] = d.partyName;  
        });  
    return partyNameDict;
}
```
SingleDistrict.tsx
```
const showPartyName = getPartyNameDict(currentDistrictResult!);
...
 {<span className="has-text-success">{showPartyName[vulnerable.winner.partyCode]}</span>}
```

..or just skip it for now and go for a better solution later.